src/shared/firewall-util-nft.c

```diff
@@ -905,31 +996,43 @@ int fw_nftables_add_local_dnat(
         if (remote_port <= 0)
                 return -EINVAL;
 
-        data[1] = htobe16(remote_port);
+        if (af == AF_INET) {
+                dlen = 8;
+                data[1] = htobe16(remote_port);
+        } else {
+                dlen = sizeof(data);
+                data[4] = htobe16(remote_port);
+        }
 
         r = sd_nfnl_message_batch_begin(ctx->nfnl, &transaction[0]);
         if (r < 0)
                 return r;
 
         tsize = 1;
         /* If a previous remote is set, remove its entry */
-        if (add && previous_remote && previous_remote->in.s_addr != remote->in.s_addr) {
-                data[0] = previous_remote->in.s_addr;
+        if (add && previous_remote && !in_addr_equal(af, previous_remote, remote)) {
+                if (af == AF_INET)
+                        data[0] = previous_remote->in.s_addr;
+                else
+                        memcpy(data, &previous_remote->in6, sizeof(previous_remote->in6));
```

Comment:
why memcpy()? simple assignment should work too

Reply:
data is an array of uint32_t; how would you do this without memcpy?

Reply:
oh, true. ignore me.